src/modules/auth/token.service.ts

```diff
@@ -70,9 +69,8 @@ export class TokenService {
     return this.refreshTokenRepository.deleteRefreshTokenById(id);
   }
 
-  async handleOriginCheck(req: Request, res: Response, next: NextFunction) {
+  async handleOriginCheck(req: Request, res: Response, next: NextFunction, service: JwtService) {
```

Comment:
Please add some documentation to explain why we are checking the origin (i.e. for csrf protection) and why we have chosen this approach

Reply:
thank you for the note, check in the next commit in docs/security folder.

Reply:
Nice, the docs file looks good. IMO, it would be better to actually put the documentation next to the class itself as JSDoc to make it more likely that it will be read an maintained. This also allows the documentation to be read as intellisense.
I also added a line at the end about why other patterns couldn't be used
```suggestion
/**
 * Our approach to prevent or at least maximum decrease chances for any CSRF attacks:
 *
 * Provide a Protection using Origin in headers and combining double check with JWT: 
 * * Origin is a request header supported in most modern browsers, indicating from where the request originated, **which cannot be modified** <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin>
 * * We will store the request Origin header in JWT token after login
 * * And with every request our middleware will check if Origin in request headers is matching with origin stored in JTW token(**hacker cannot modify JWT as we are signing it with our secret in backend**)
 * * For node requests we will check if origin in request headers is undefined. 
 * * IF ORIGIN IS SAME OR UNDEFINED: everything is fine
 * * IF NO: throw an exception as the origin could be from a hacker's website.
 * * More info(check approach 2, **origin headers**): <https://security.stackexchange.com/questions/203890/how-to-implement-csrf-protection-with-a-cross-origin-request-cors/203910#203910>
 *
 * A pattern such as the double cookie submit pattern cannot be used because the cache-server does not share an origin with it's clients.
 */
  async handleOriginCheck(req: Request, res: Response, next: NextFunction, service: JwtService) {
```